pkg/commands/git.go

```diff
@@ -73,30 +72,15 @@ func NewGitCommand(
 	// to have forward slashes internally
 	currentPath = filepath.ToSlash(currentPath)
 
-	gitDir := env.GetGitDirEnv()
-	if gitDir != "" {
-		// we've been given the git directory explicitly so no need to navigate to it
-		_, err := cmn.Fs.Stat(gitDir)
-		if err != nil {
-			return nil, utils.WrapError(err)
-		}
-	} else {
-		// we haven't been given the git dir explicitly so we assume it's in the current working directory as `.git/` (or an ancestor directory)
-
-		rootDirectory, err := findWorktreeRoot(cmn.Fs, currentPath)
-		if err != nil {
-			return nil, utils.WrapError(err)
-		}
-		currentPath = rootDirectory
-		err = os.Chdir(rootDirectory)
-		if err != nil {
-			return nil, utils.WrapError(err)
-		}
+	repoPaths, err := git_commands.GetRepoPaths(osCommand.Cmd, currentPath)
+	if err != nil {
+		return nil, errors.Errorf("Error getting repo paths: %v", err)
 	}
+	currentPath = repoPaths.WorktreePath()
 
-	repoPaths, err := git_commands.GetRepoPaths(cmn.Fs, currentPath)
+	err = os.Chdir(currentPath)
```

Comment:
This code is a little convoluted: we start with currentPath being defined based on the CWD, then we pass that into GetRepoPaths and all that does with it is store it on the RepoPaths struct, then back in this function we chdir to worktreePath but now have a stale currentPath on the RepoPaths struct (given that currentPath is meant to represent the CWD).

Given that we're not actually using RepoPaths's currentPath anywhere (given we don't support opening lazygit in a subdirectory of a worktree), I think we should just remove it from the struct entirely. When we implement that feature, we can add it back, but right now it only causes confusion.

As such we'll need no mention of `currentPath` in GetRepoPaths, or in this function. We can just do:
```go
err = os.Chdir(repoPaths.WorktreePath())
```

I'm happy to continue explicitly passing the worktree root path to diff commands though given that we will want to retain that logic when we do support opening lazygit in worktree subdirectories.

Reply:
Sounds good. Also, I think there's a good discussion to be had about what you intended for the UX. AFAICT, this PR now, really and truly, fixes all issues with running lazygit from an arbitrary subdirectory – that's a fair bit of what the fixes and integ tests around symlinks/submodules/bare repos/etc. tackle. The worktree side of that should now be rock solid. BUT – that doesn't show things in path-relative sense ala `git status`, always from repo root.  I'm not clear that relative paths would be that useful for lazygit, but I'll bet that there are users with big/deep repos who have pain points that UX improvements could help a lot, and subdir/tree-narrowing ideas may help there.